plugins/org.jboss.tools.livereload.core/src/org/jboss/tools/livereload/core/internal/util/WSTUtils.java

```diff
@@ -186,11 +190,33 @@ public static IServer extractServer(final String browserLocation, final List<ISe
 	}
 
 	/**
+	 * Retrieves the hostname for the given server, with support for specific types of servers, such as OpenShift 3.
+	 * @param server the server to analyze
+	 * @return the server hostname
+	 * @throws ExecutionException if something went wrong
+	 */
+	public static String getWebHost(final IServer server) throws ExecutionException {
+		final String serverType = server.getServerType().getId();
+		final Object adapter = server.loadAdapter(ServerDelegate.class, null);
+		if (serverType.equals(OPENSHIFT_SERVER_TYPE)) {
+			if(adapter instanceof IExtendedPropertiesProvider) {
+				final IExtendedPropertiesProvider propertiesProvider = (IExtendedPropertiesProvider) adapter;
+				try {
+					final URL welcomePageURL = new URL(propertiesProvider.getExtendedProperties().getWelcomePageUrl());
```

Comment:
Case null should be checked and nothing should be done, because implementation of getWelcomePageUrl() calls message dialog informing user that there is nothing to show. Returning server.getHost() in this case will create problems that will be unfriendly logged after user already was kindly informed of the problem. So there should be a way in the case of null to silently quit like a gentleman does.